force-app/main/schema/classes/user/UserMapper.cls

```diff
@@ -0,0 +1,121 @@
+    public static UserMapper getInstance() {
+        if (instance == null) {
+            instance = new UserMapper();
+        }
+
+        return instance;
+    }
+
+    /************************************************************************************************************
+     * @description Retrieve a UserModel representing a user and its basic fields
+     * @param userId The Id of the user to retrieve
+     * @return A UserModel
+     ************************************************************************************************************/
+    public virtual UserModel getUserModelById(Id userId) {
+        User userRetrieved = [
+            SELECT Id, Name, Email, IsActive, SmallPhotoUrl
+            FROM User
+            WHERE Id = :userId
+            WITH SECURITY_ENFORCED
+        ];
+
+        return new UserModel(
+            userRetrieved.Id,
+            userRetrieved.Name,
+            userRetrieved.IsActive,
+            userRetrieved.Email,
+            userRetrieved.SmallPhotoUrl
+        );
+    }
+
+    /************************************************************************************************************
+     * @description Retrieve a list of UserModel representing a user and their basic fields
+     * @param userNameMatch The string to match user names to
+     * @param recordLimit The number of records to limit the query to
+     * @return A list of User Models with names that match the search string
+     ************************************************************************************************************/
+    public virtual List<UserModel> getUserModelsLikeName(String userNameMatch, Integer recordLimit) {
+        String sanitizedUserNameMatchSearch = '%' + String.escapeSingleQuotes(userNameMatch) + '%';
+
+        List<User> usersRetrievedList = [
+            SELECT Id, Name, Email, IsActive, SmallPhotoUrl
+            FROM User
+            WHERE Name LIKE :sanitizedUserNameMatchSearch
+            WITH SECURITY_ENFORCED
+            LIMIT :recordLimit
+        ];
+
+        List<UserModel> userModelList = new List<UserModel>();
+
+        for (User userRetrieved : usersRetrievedList) {
+            userModelList.add(
+                new UserModel(
+                    userRetrieved.Id,
+                    userRetrieved.Name,
+                    userRetrieved.IsActive,
+                    userRetrieved.Email,
+                    userRetrieved.SmallPhotoUrl
+                )
+            );
+        }
+        return userModelList;
+    }
+
+    /************************************************************************************************************
+     * @description Retrieve the label for the user object
+     * @return The label for the user object
+     ************************************************************************************************************/
+    public virtual String getModelObjectLabel() {
```

Comment:
I dont think we need this method if we really need to just call UTIL_Describe.getObjectDescribe.

Reply:
This is important from a mocking perspective and for us to have the flexibility to add specific labels if we need to.